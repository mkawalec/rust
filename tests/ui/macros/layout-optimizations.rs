#![feature(pattern_types)]
#![allow(incomplete_features)]

fn main() {
    type Wrong = layout_optimizations!(u32, 1 | 2);
    //~^ Pattern types only support range expressions
    type Percent = layout_optimizations!(u32, 0..=100);
    //~^ pattern types are unstable
    let blah: layout_optimizations!(u32, 123..) = 256;
}
