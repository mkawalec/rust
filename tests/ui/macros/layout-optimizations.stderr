error: Pattern types only support range expressions
  --> $DIR/layout-optimizations.rs:5:18
   |
LL |     type Wrong = layout_optimizations!(u32, 1 | 2);
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-^
   |                                                 |
   |                                                 help: try supplying a range expression instead

error[E0308]: mismatched types
  --> $DIR/layout-optimizations.rs:9:51
   |
LL |     let blah: layout_optimizations!(u32, 123..) = 256;
   |               ---------------------------------   ^^^ expected `(u32) is 123..=`, found integer
   |               |
   |               expected due to this
   |
   = note: expected pattern type `(u32) is 123..=`
                      found type `{integer}`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
